Configuration issue on version 0.10.8 of oss-licenses-plugin

**Describe the bug**

Updated versions this morning and saw this error when trying to run `./gradlew clean`

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':app:debugOssLicensesCleanUp' (type 'LicensesCleanUpTask').
  - In plugin 'com.google.android.gms.oss-licenses-plugin' type 'com.google.android.gms.oss.licenses.plugin.LicensesCleanUpTask' property 'generatedDirectory' specifies directory '/Users/.../app/build/generated/third_par.

    Reason: An input file was expected to be present but it doesn't exist.

    Possible solutions:
      1. Make sure the directory exists before the task is called.
      2. Make sure that the task which produces the directory is declared as an input.

    For more information, please refer to https://docs.gradle.org/8.13/userguide/validation_problems.html#input_file_does_not_exist in the Gradle documentation.
```


**To Reproduce**
Steps to reproduce the behavior:
1. Run `./gradlew clean`

**Expected behavior**
`./gradlew clean` should work

**Desktop (please complete the following information):**
 - Gradle version: 8.13, AGP version 8.11
 - Android Studio version: Android Studio Narwhal Feature Drop | 2025.1.2 Patch 1
 - Plugin name and version: OSS licenses 0.10.8

--- a/oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/LicensesCleanUpTask.groovy
+++ b/oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/LicensesCleanUpTask.groovy
@@ -20,6 +20,7 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.tasks.InputDirectory
 import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.SkipWhenEmpty
 import org.gradle.api.tasks.TaskAction
 import org.gradle.work.DisableCachingByDefault
 
@@ -29,6 +30,7 @@ import org.gradle.work.DisableCachingByDefault
 @DisableCachingByDefault(because = "Local deletion operation")
 abstract class LicensesCleanUpTask extends DefaultTask {
     @Optional
+    @SkipWhenEmpty
     @InputDirectory
     abstract DirectoryProperty getGeneratedDirectory()
 

--- a/oss-licenses-plugin/src/test/java/com/google/android/gms/oss/licenses/plugin/EndToEndTest.kt
+++ b/oss-licenses-plugin/src/test/java/com/google/android/gms/oss/licenses/plugin/EndToEndTest.kt
@@ -3,6 +3,7 @@ package com.google.android.gms.oss.licenses.plugin
 import org.gradle.testkit.runner.GradleRunner
 import org.gradle.testkit.runner.TaskOutcome
 import org.junit.Assert
+import org.junit.Before
 import org.junit.Test
 import org.junit.rules.TemporaryFolder
 import org.junit.Rule
@@ -30,9 +31,11 @@ class EndToEndTest(private val agpVersion: String, private val gradleVersion: St
 
     private fun isBuiltInKotlinEnabled() = agpVersion.startsWith("9.")
 
-    @Test
-    fun basic() {
-        val projectDir = tempDirectory.newFolder("basic")
+    private lateinit var projectDir: File
+
+    @Before
+    fun setup() {
+        projectDir = tempDirectory.newFolder("basic")
         File(projectDir, "build.gradle").writeText(
             """
             plugins {
@@ -70,6 +73,10 @@ class EndToEndTest(private val agpVersion: String, private val gradleVersion: St
             }
             """.trimIndent()
         )
+    }
+
+    @Test
+    fun basic() {
         val result = GradleRunner.create()
             .withProjectDir(projectDir)
             .withGradleVersion(gradleVersion)
@@ -84,7 +91,26 @@ class EndToEndTest(private val agpVersion: String, private val gradleVersion: St
         val metadata =
             File(projectDir, "build/generated/resources/releaseOssLicensesTask/raw/third_party_license_metadata")
         Assert.assertEquals(expectedContents(isBuiltInKotlinEnabled()), metadata.readText())
+
+        val cleanResult = GradleRunner.create()
+            .withProjectDir(projectDir)
+            .withGradleVersion(gradleVersion)
+            .withArguments("clean", "-s")
+            .build()
+        Assert.assertFalse(File(projectDir, "build").exists())
+        Assert.assertEquals(cleanResult.task(":clean")!!.outcome, TaskOutcome.SUCCESS)
     }
+
+    @Test
+    fun clean() {
+        val result = GradleRunner.create()
+            .withProjectDir(projectDir)
+            .withGradleVersion(gradleVersion)
+            .withArguments("clean", "-s")
+            .build()
+        Assert.assertEquals(result.task(":clean")!!.outcome, TaskOutcome.UP_TO_DATE)
+    }
+
 }
 
 private fun expectedDependenciesJson(builtInKotlinEnabled: Boolean) = """[

